### PR TITLE
feat(backend): Increase maximum possible visualization size

### PR DESCRIPTION
**Pull request by hbrylkowski**
**Description of your changes:**

I had some issues with bigger visualizations, I noticed that there is already a [document](https://github.com/kubeflow/pipelines/blob/master/backend/src/apiserver/visualization/README.md) wihich describes how to fix this issue. I made PR with code from the said document and removed section about the issue since it's no longer applicable.

**Checklist:**
- [x] The title for your pull request (PR) should follow our title convention. [Learn more about the pull request title convention used in this repository](https://github.com/kubeflow/pipelines/blob/master/CONTRIBUTING.md#pull-request-title-convention). 

   PR titles examples:
    * `fix(frontend): fixes empty page. Fixes #1234`
       Use `fix` to indicate that this PR fixes a bug.
    * `feat(backend): configurable service account. Fixes #1234, fixes #1235`
       Use `feat` to indicate that this PR adds a new feature. 
    * `chore: set up changelog generation tools`
       Use `chore` to indicate that this PR makes some changes that users don't need to know.
    * `test: fix CI failure. Part of #1234`
        Use `part of` to indicate that a PR is working on an issue, but shouldn't close the issue when merged.

- [x] Do you want this pull request (PR) cherry-picked into the current release branch?
    
    If yes, use one of the following options:
  
    * **(Recommended.)** Ask the PR approver to add the `cherrypick-approved` label to this PR. The release manager adds this PR to the release branch in a batch update.
    *  After this PR is merged, create a cherry-pick PR to add these changes to the release branch. (For more information about creating a cherry-pick PR, see the [Kubeflow Pipelines release guide](https://github.com/kubeflow/pipelines/blob/master/RELEASE.md#option--git-cherry-pick).)

Is it possible to add it to 1.0 release?

### CODE REVIEW

**Comment by Ark-kun** [comment]
/ok-to-test

Thank you for the improvement.

Can you please tell us how you use visualizations?
We have some plans to rework visualizations as components that turn data to HTML or Markdown. https://github.com/kubeflow/pipelines/issues/3510

What kind of visualizations have you created? Maybe you can create components based on them.

**Comment by hbrylkowski** [comment]
Hi, Thank you for the input.

Right now I'm using tensorflow extended and all visualizations comes from there. My specific issue arised when I had dataset with encoded images and StatisticsGen component generated HTML which contained few encoded images and it resulted in visualization which was bigger than default 4MB. As for the visualisation beeing separate component - I do not have enough experience with kubeflow yet to form opinions about it.

For the PR itself I wonder - maybe it would be better idea to pass maximum possible visualization size in environment var?

File: backend/src/apiserver/main.go [hunk 1]
```diff
@@ -149,6 +149,12 @@ func registerHttpHandlerFromEndpoint(handler RegisterHttpHandlerFromEndpoint, se
 	endpoint := "localhost" + *rpcPortFlag
 	opts := []grpc.DialOption{grpc.WithInsecure()}
 
+	// Only change the maxCallRecvMesSize if it is for visualizations
+	if serviceName == "Visualization" {
+		maxCallRecvMsgSize := 50 * 1024 * 1024
```
Comment (jingzhang36) [inline backend/src/apiserver/main.go:154]: Would you mind add a little bit more comments to this? E.g, default message size is 4MB and also a comment to indicate that the 50 * 1024 * 1024 is how many MB . Thanks!
Other than that /lgtm

Comment (hbrylkowski) [reply backend/src/apiserver/main.go]: Done, sorry for late response

**Comment by Bobgy** [comment]
Later already updated in https://github.com/Bobgy/pipelines/blob/6c326ad8eab30387fc18f6b5e8b09bea6d6c52e5/backend/src/apiserver/main.go#L162

**Pull request closed without merging**